### ISSUE TITLE
Getting undefined when creating new record

### ISSUE THREAD

**Issue by jlcarvalho** [fortunejs/fortune#133]
Fortune returns:

```
TypeError: Cannot set property \'email\' of undefined
    at /var/www/node_modules/fortune/dist/lib/dispatch/create.js:259:49
    at run (/var/www/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:89:39)
    at /var/www/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:100:28
    at flush (/var/www/node_modules/babel-runtime/node_modules/core-js/library/modules/$.microtask.js:17:13)
    at process._tickCallback (node.js:415:13)
```

The code runs OK locally, but in my remote Digital Ocean VPS i get this exception.

```js
store.defineType('pessoa', {
    criadoEm: {type: Date},
    modificadoEm: {type: Date},
    nome: { type: String },
    email: { link: 'email', inverse: 'pessoas' },
    nascimento: { type: Date },
    usuario: { link: 'usuario', inverse: 'pessoa' },
    certificados: { link: 'certificado', inverse: 'pessoa', isArray: true }
})

store.defineType('email', {
    criadoEm: {type: Date},
    modificadoEm: {type: Date},
    nome: { type: String },
    pessoas: { link: 'pessoa', inverse: 'email', isArray: true }
})

store.transformInput('pessoa', (context, record, update) => {
    const { request: { method, type, meta } } = context

    if (method === methods.create) {
        return store.request({
            method: methods.find,
            type: 'email',
            options: {
                match: {
                    nome: record.email
                }
            }
        }).then(function(res){
            return res.payload.count ?
                Promise.resolve(res) :
                store.request({
                    method: methods.create,
                    type: 'email',  // not plurialized
                    payload: [{    // push your data into an array
                        nome: record.email
                    }]
                })
        })
        .then(function (res) {
            console.dir(res.payload[0])
            record.email = res.payload[0].id
            return record
        })
    }
    return update
})
```

**Comment by 0x8890** [comment]
Need to know more to debug:

- Fortune version
- Node version
- Create payload
- Adapter

**Comment by jlcarvalho** [comment]
VPS
- Fortune version: 1.3.11
- Node: 4.0
- OS: Ubuntu 15.04 LTS
- Adapter: MongoDB

Local
- Fortune version: 1.3.11
- Node: 4.0
- OS: OSX 10.10.5
- Adapter: MongoDB

The request is on the `Usuario` type, which creates a record on `Pessoa` type.
Request payload:
```json
{"@graph":
  [{
    "username":"jlcarvalho",
    "password":"jean2311",
    "nome":"Jean Lucas de Carvalho",
    "email":"jeanlucasdecarvalho@gmail.com",
    "@type":"Usuario"
  }],
  "@context": {
    "@vocab":"http://schema.org/",
    "µ":"http://micro-api.org/"
  }
}
```

```js
const [ iterations, keyLength, saltLength ] =
    [ Math.pow(2, 15), Math.pow(2, 9), Math.pow(2, 6) ]

function passwordCheck (password, key, salt) {
    return new Promise((resolve, reject) => crypto.pbkdf2(
        password, salt, iterations, keyLength, (error, buffer) =>
            error ? reject(error) : key.equals(buffer) ? resolve() : reject()))
}

function generateSalt () {
    return new Promise((resolve, reject) =>
        crypto.randomBytes(saltLength, (error, buffer) =>
            error ? reject(error) : resolve(buffer)))
}

function generateKey (password, salt) {
    return new Promise((resolve, reject) =>
        crypto.pbkdf2(password, salt, iterations, keyLength, (error, buffer) =>
            error ? reject(error) : resolve(buffer)))
}

store.defineType('usuario', {
    criadoEm: {type: Date},
    modificadoEm: {type: Date},
    username: { type: String },
    key: { type: Buffer },
    salt: { type: Buffer },
    admin: { type: Boolean },
    pessoa: { link: 'pessoa', inverse: 'usuario' },
    modelos: { link: 'modelo', inverse: 'usuarios', isArray: true }
})

store.transformInput('usuario', (context, record, update) => {
    const { request: { method, type, meta } } = context
    let { key, salt, password } = record

    if (method === methods.create && !password)
        throw new errors.BadRequestError(`Senha não informada.`)

    return method !== methods.create ? passwordCheck(
        new Buffer(meta['authorization'] || '', 'base64').toString(),
        key, salt.toString()) : Promise.resolve()

        .catch(() => {
            throw new errors.UnauthorizedError(`Senha incorreta.`)
        })

        .then(() => {
            if (method === methods.delete) return null
            if (!password) password = update.replace.password

            return generateSalt()
                .then(buffer => {
                    salt = buffer
                    return generateKey(password, salt.toString())
                })
                .then(buffer => {
                    key = buffer
                    if (method === methods.create) {
                        return store.request({
                            method: methods.create,
                            type: 'pessoa',  // not plurialized
                            payload: [{    // push your data into an array
                                nome: record.nome,
                                email: record.email
                            }]
                        })
                        .then(function (res) {
                            record.key = key
                            record.salt = salt
                            record.pessoa = res.payload[0].id
                            return record
                        })
                        .catch(function(e){
                            console.dir(e)
                                console.dir(e.stack)
                        })

                    }
                    update.replace = { key, salt }

                    return update
                })
        })
})
```

**Comment by jlcarvalho** [comment]
The code running on VPS is transpiled using gulp-babel 5.2.1.

**Comment by 0x8890** [comment]
Do you know which part of the application code causes the error?

**Comment by 0x8890** [comment]
I released `1.3.12`, it may possibly solve this problem?

**Comment by jlcarvalho** [comment]
Solved. Thanks.
